src/util/parented_ptr.h

```diff
@@ -88,34 +98,41 @@ inline parented_ptr<T> make_parented(Args&&... args) {
     return parented_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+template<typename T>
+inline parented_ptr<T> to_parented(std::unique_ptr<T>& u) noexcept {
+    // the DEBUG_ASSERT in the parented_ptr constructor will catch cases where the unique_ptr should
```

Comment:
There aren't any debug assertions in the constructor, only in the destructor?

Reply:
QObjects managed by a unique_ptr should never have a parent to avoid a double free. Instead the parent should be set when releasing it from the unique_ptr.

```
parented_ptr<T> to_parented(std::unique_ptr<T>& u, QObject* parent) noexcept {
    DEBUG_ASSERT(u);
    DEBUG_ASSERT(!u->parent());
    DEBUG_ASSERT(parent);
    auto p = u.release();
    p->setParent(parent);
    return parented_ptr<T>{p};
}
```

The constructor should assert that the wrapped QObject actually has a parent. In the destructor or when abandoning m_ptr we should do the same, because this wrapper has no control over the wrapper QObject and its parent/child relations during the lifetime.

Reply:
About the comment: Right, my comment in the code is wrong.

About your suggestion:
That would still leave the problem of having two different pointers that might currently point to the object, depending on the context.

Also, sometimes there are other methods used (different from `child->setParent()`) to add a child. That might for example be a method on the parent where the (to-become) child is given together with some additional args.

With my PR, one could then write:

```
parent->someMethodThatAddsAChild(make_parented(childCtorArgs), moreArgs);
```

In that code, the `parented_ptr` is destructed after the function call, so the debug assertion succeeds.

Reply:
Agreed, this will improve usability. Any inconsistencies will eventually be caught by the debug assertion in the destructor.